File: lv1-library-loan/typescript/src/application/books/book-repository-errors.ts
import type { BookId } from "../../domain/books/book-id.js";
import { ResultError } from "../../results/result-error.js";

export const BookRepositoryErrors = {
  notFound: (bookId: BookId): ResultError => new ResultError("Book.NotFound", `本(ID: ${bookId})が見つかりません。`),
};

File: lv1-library-loan/csharp/LibraryLoan.Application/Books/BookRepositoryErrors.cs
using LibraryLoan.Domain.Books;
using LibraryLoan.Results;

namespace LibraryLoan.Application.Books;

public static class BookRepositoryErrors
{
    public static Error NotFound(BookId bookId) => new(
        "Book.NotFound",
        $"本(ID: {bookId})が見つかりません。");
}
